### PR TITLE
Rep 4029 - Add stripped links to xml response

### PR DESCRIPTION
**Pull request by JoelRizner**

### CODE REVIEW

File: repose-aggregator/components/filters/uri-stripper-filter/src/main/resources/META-INF/schema/examples/uri-stripper.cfg.xml [hunk 1]
```diff
@@ -3,8 +3,19 @@
 <uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="true" token-index="1">
     <link-resource uri-path-regex=".*" http-methods="ALL">
         <response>
+
```
Comment (wdschei) [inline repose-aggregator/components/filters/uri-stripper-filter/src/main/resources/META-INF/schema/examples/uri-stripper.cfg.xml]: Why the extra white space?

File: repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala [hunk 2]
```diff
@@ -171,7 +192,7 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
     Option(contentType) match {
       case Some(ct) if ct.toLowerCase.contains("json") => resource.getJson.toList
       // todo: return xpath when xml is supported
```
Comment (wdschei) [inline repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala]: Should this `TODO` be removed now?

File: repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala [hunk 3]
```diff
@@ -231,6 +252,20 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
...
+    val transformerFactory = TransformerFactory.newInstance()
+    val xsltTransformer = transformerFactory.newTransformer(xsltSource)
+    xsltTransformer.setParameter(parameterName, parameterValue)
+
+    tryResponseXml map { responseXml =>
+      Try(responseXml.transform)
+
+
+
+
```
Comment (wdschei) [inline repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala]: Why all the extra white space?

Comment (JoelRizner) [inline repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala]: yeah.. that was where i finished friday so i pushed, in retrospect i should have just stashed the unfinished changes.

**Comment by wdschei** [comment]
The following test seems to be missing from the `UriStripperSchemaTest.scala`
```scala
it(s"should successfully validate if only json is present") {
  val config =
  s"""<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="true" token-index="1">
            |    <link-resource>
            |        <request>
            |            <json>$$.service.link</json>
            |        </request>
            |        <response>
            |            <json>$$.service.link</json>
            |        </response>
            |    </link-resource>
            |</uri-stripper>""".stripMargin

  validator.validateConfigString(config)
}
```

**Comment by adrianjgeorge** [comment]
retest this please

**Comment by JoelRizner** [comment]
Could this failure be because i added saxon HE in to the properties? The failure is in the saxonEE functionality tests in the translation filter. I added saxon HE because Jorge was saying EE needed a license so i wasn't sure there..

**Comment by wdschei** [comment]
That is possible. I was wondering why you needed to add that.

**Comment by wdschei** [comment]
The EE only needs a license if you do EE stuff. If you are not doing EE stuff then it should be good.

**Comment by adrianjgeorge** [comment]
Yeah, the ee tests are losing the classpath war. he is  showing up first

File: repose-aggregator/components/filters/uri-stripper-filter/build.gradle [hunk 4]
```diff
@@ -6,6 +6,9 @@ dependencies {
     compile "com.typesafe.scala-logging:scala-logging-slf4j_$scalaMajDotMin"
     compile "com.typesafe.play:play-json_$scalaMajDotMin"
     compile "io.gatling:jsonpath_$scalaMajDotMin"
+    compile "org.scala-lang.modules:scala-xml_$scalaMajDotMin"
+    compile "com.rackspace.cloud.api:wadl-tools_$scalaMajDotMin"
```
Comment (wdschei) [inline repose-aggregator/components/filters/uri-stripper-filter/build.gradle]: Is Repose actually using WADL Tools now?

File: repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala [hunk 5]
```diff
@@ -127,21 +124,81 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
...
+    }
+  }
+
+  private def handleXmlResponseLinks(wrappedResponse: HttpServletResponseWrapper, previousToken: Option[String], nextToken: Option[String], token: Option[String], applicableLinkPaths: List[LinkPath]): Unit = {
+    try {
+      val result = applicableLinkPaths.foldLeft(wrappedResponse.getOutputStreamAsInputStream) { (in: InputStream, linkPath: LinkPath) =>
+        val out = new ByteArrayOutputStream()
+        val transformer = templateMap.get(linkPath).get.newTransformer
+        transformer.asInstanceOf[Controller].addLogErrorListener
+        transformer.setParameter("removedToken", token.getOrElse("")) //todo make optional
```
Comment (wdschei) [inline repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala]: What is the plan for making this an Optional?

**Comment by JoelRizner** [comment]
We are using WADL tools for the transformation, provided by Jorge

File: repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala (hunk 5 above)
Comment (JoelRizner) [inline repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala]: I can remove that todo, it was from a while back. We agreed to pass an empty string instead and handle that logic in the xslt

**Comment by wdschei** [comment]
This looks good. Way to knock out the Marshaller!!!

**Pull request merged**